### DISCUSSION TITLE
v1.17 beta1

### DISCUSSION THREAD

**Discussion by duracelltomi** [duracelltomi/gtm4wp#265] [General]
* Added: pagePostTerms data layer variable will now also include a meta key with post meta values that does not start with the _ character. This should allow to utilize custom fields added by plugins like Advanced Custom Fields.
* Added: new filter: gtm4wp_post_meta_in_datalayer to limit which post meta should be present with with the previously added feature.
* Added: ability to use a custom path for your server side GTM container. This makes GTM4WP compatible with services like stape.io
* Added: new advanced option to disable browser console.log() messages on frontend
* Added: new filter: gtm4wp_purchase_datalayer to be able to alter data layer content for the purchase data
* Added: new filter: gtm4wp_datalayer_on_pageload to be able to alter data layer content generated during page load
* Added: customerBillingState and customerShippingState on WooCommerce order received page, thanks [massimo-maimeri](https://github.com/massimo-maimeri)
* Added: High Performance Order Storage (HPOS) compatibility for WooCommerce
* Updated: WooCommerce integration will also check target attitbute of product links while firing select_item event, thanks [robklo])(https://github.com/robklo)
* Updated: YouTube tracking will not load on pages not containing a YouTube block or a YouTube iframe
* Fixed: add_to_cart event on grouped products not working when SKU is selected as product ID in plugin options.
* Fixed: more reliable new_customer flag working with guest orders of existing customers, thanks [morvy](https://github.com/morvy)

<hr /><em>This discussion was created from the release <a href='https://github.com/duracelltomi/gtm4wp/releases/tag/1.17beta1'>v1.17 beta1</a>.</em>

**Comment by jumbotaylor92** [comment]
@duracelltomi tamás geiger

I am not a developer but I use this plugin on a daily basis. So please excuse my technical terms here. 
I  have been testing the beta version on the staging site of mine, and the repeating issue that I am facing is the "purchase event#"
the data layer is been pushed sometimes but irregularly and the same has not been passed to GA4 as well.

gtag has stopped sending the purchase event. this is only regarding the purchase event. 

Please check 

Thank you. 
JP

**Comment by duracelltomi** [comment]
Hi,

Were you able to find a fixed path that leads to a missing purchase event?
Can you post the URL of your staging site so that I can place a test order?

**Comment by jumbotaylor92** [comment]
@duracelltomi 
https://www.staging6.harveymaria.com/

It's the same issue with the live site as well.  This is the live site: http://harveymaria.com/

Thank you for looking into it.

Best
JP

**Comment by duracelltomi** [comment]
I did a test and purchase data was there in the data layer:

```
{
	event: "purchase",
	ecommerce: {
		currency: "GBP",
		items: [
			{
				item_id: 1064120,
				item_name: "Walbrook Peyton Blue",
				item_brand: "All Flooring Styles",
				price: 51.1,
				item_category: "LVT Flooring",
				quantity: 1,
				google_business_vertical: "retail",
				id: "1064120"
			}
		],
		transaction_id: "1070352",
		affiliation: "",
		value: 89.1,
		tax: 0,
		shipping: 38,
		coupon: ""
	}
}

```

However, the page reloaded itself for some reason the double tracking protection turned on. The second page load of the order received page did not include the purhcase data for that reason.

![image](https://user-images.githubusercontent.com/7515949/235193942-740ae9e3-e05e-4a4c-85f1-6ac3101c01a7.png)

**Comment by RahiReja** [comment]
@duracelltomi

![image](https://user-images.githubusercontent.com/16631310/235404217-6ceaa3f5-6764-41ca-91ec-2303bea1448e.png)

Can you tell me where I should find those features in the new version because I couldn't find that?

**Reply by duracelltomi** [reply]
These are WordPress filters. You need to create PHP codes in order to utilize those additions: https://developer.wordpress.org/plugins/hooks/filters/

**Reply by RahiReja** [reply]
Can you give me an example code of how to enable those or guide me to any doc where I can find it in more detail?

**Comment by maxwo007** [comment]
I have been using the beta for two weeks now. I have not run into any problems. When would you release this via wordpress?

**Reply by duracelltomi** [reply]
It has been just released here on Github, I will upload it to WP.org tomorrow as I will be not at my desk in the next 24 hours
